### PR TITLE
[CI/CD] Improvements

### LINKED ISSUE #841: [CI/CD] Improvements
The following improvements should be adapted to the CI/CD system so that it is more helpful for development:

- [x] Splitting up the build task so that it is immediately obvious where the error is coming from.
  - [x] `fbs-core.api` Action
  - [x] `fbs-core.web` Action
  - [x] `fbs-runner.checker` Action
  - [x] Improve PR Report Title
  - [x] `fbs-sql.checker` Action
- [x] Build dockerfiles on every PR
- [x] Docker Hub Publish improvements
  - [x] use `docker/login-action`
  - [x] use `environment`
  - [x] Hide on PRs
  - [x] Warn if docker build fail
- [x] Improvement of the Open API Vallidation (The current check does not find all errors) -> may use `npx  @redocly/cli@latest lint <path>`
- [x] Code coverage Verbesserungen
  - [x] Split coverage for every module
  - [x] Switch to `codecov`
- [x] Add `environment` for the Testsystem Deplyoment 
- [x] Run only single Testsystem Deploy Action
- [x] Add annotations for liniting errors (may use [actions Toolkit](https://github.com/actions/toolkit/tree/main/packages/core#annotations) or [problem-matchers](https://github.com/actions/runner/blob/main/docs/adrs/0276-problem-matchers.md))
- [x] #835
- [x] Fix `node 12` deprcated Warnung ⚠️

### PR DESCRIPTION
**Pull request by mxsph**
resolves #841

### CODE REVIEW

File: modules/fbs-runner/checker/build.gradle [hunk 1]
```diff
@@ -46,3 +45,16 @@ jar {
...
+    additionalSourceDirs.setFrom files('src/main/scala')
+    sourceDirectories.setFrom files('src/main/scala')
+    classDirectories.setFrom files('build/classes/scala/main')
+    executionData.setFrom files('build/jacoco/test.exec')
+
+    reports {
+        xml.enabled true
+        html.enabled true
+    }
+}
```
Comment (Zitrone44) [inline modules/fbs-runner/checker/build.gradle:60]:
```suggestion
}

```

File: modules/fbs-core/api/build.gradle [hunk 2]
```diff
@@ -87,3 +85,16 @@ dependencies {
...
+    additionalSourceDirs.setFrom files('src/main/scala')
+    sourceDirectories.setFrom files('src/main/scala')
+    classDirectories.setFrom files('build/classes/scala/main')
+    executionData.setFrom files('build/jacoco/test.exec')
+
+    reports {
+        xml.enabled true
+        html.enabled true
+    }
+}
```
Comment (Zitrone44) [inline modules/fbs-core/api/build.gradle:100]:
```suggestion
}

```

File: scripts/ci/docker-deploy.sh [hunk 3]
```diff
@@ -17,21 +17,25 @@ function dockerPush(){
...
-if [[ "dev" == "$branch" ]]
-then
-    dockerPush dev-latest
-elif [[ "main" == "$branch" ]]
-then
-    dockerPush latest
-else
-    dockerPush $branch
-fi
+dockerPush $tag
```
Comment (Zitrone44) [inline scripts/ci/docker-deploy.sh:41]:
```suggestion
dockerPush $tag

```

File: scripts/ci/validate-api-docs/validate-api-docs.sh [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+set -e
+
+npm --prefix scripts/ci/validate-api-docs install
+npm --prefix scripts/ci/validate-api-docs run lint
```
Comment (Zitrone44) [inline scripts/ci/validate-api-docs/validate-api-docs.sh:6]:
```suggestion
npm --prefix scripts/ci/validate-api-docs run lint

```

**Review by Zitrone44 — CHANGES_REQUESTED** [review]
Looks very good :+1:

Only some end of file new line issues.

**Review by Zitrone44 — APPROVED** [review]

**Review by Medienkunzt — APPROVED** [review]

**Pull request merged**